File: fetch_weather.js
#!/usr/bin/env node

const https = require('https');
const fs = require('fs');

const URL = 'https://wttr.in/Toronto?format=j1';
const OUTPUT_FILE = 'weather.json';
const INTERVAL = 300 * 1000; // 5 minutes in ms

function fetchWeather() {
  https.get(URL, (res) => {
    let data = '';
    res.on('data', chunk => data += chunk);
    res.on('end', () => {
      try {
        const json = JSON.parse(data);
        fs.writeFile(OUTPUT_FILE, JSON.stringify(json, null, 2), (err) => {
          if (err) {
            console.error('Error writing file:', err);
          } else {
            console.log(`Updated ${OUTPUT_FILE}`);
          }
        });
      } catch (e) {
        console.error('Error parsing weather JSON:', e);
      }
    });
  }).on('error', (e) => {
    console.error('Error fetching weather:', e);
  });
}

function main() {
  fetchWeather();
  setInterval(fetchWeather, INTERVAL);
}

main();

File: fetch_weather.py
#!/bin/bash
'''exec' python3 "$0" "$@"
' '''
import time
import json
import urllib.request

URL = "https://wttr.in/Toronto?format=j1"
OUTPUT_FILE = "weather.json"
INTERVAL = 300  # 5 minutes

def fetch_weather():
    try:
        with urllib.request.urlopen(URL, timeout=10) as response:
            data = json.load(response)
        with open(OUTPUT_FILE, "w", encoding="utf-8") as f:
            json.dump(data, f, indent=2)
        print(f"Updated {OUTPUT_FILE}")
    except Exception as e:
        print(f"Error fetching weather: {e}")

def main():
    while True:
        fetch_weather()
        time.sleep(INTERVAL)

if __name__ == "__main__":
    main()
